client/src/components/ApplicationsEntriesSelector/ApplicationsEntriesSelector.tsx

```diff
@@ -0,0 +1,137 @@
+import './ApplicationsEntriesSelector.scss';
+import React, {useState, useEffect} from 'react';
+import Table, {TableColumn} from 'components/Table/Table';
+import Checkbox from 'components/Checkbox/Checkbox';
+import {ManagmentServiceApiInstance} from 'api/managment-service';
+import LinkComponent from 'components/LinkComponent/LinkComponent';
+import {ApplicationDataRow} from 'pages/Report/ApplicationSection/ApplicationSection';
+import ActionButton, {ActionButtonColor} from 'components/ActionButton/ActionButton.tsx';
+
+interface ApplicationsEntriesSelectorProps {
+    selectedApplications: ApplicationDataRow[];
+    setSelectedApplications: React.Dispatch<React.SetStateAction<ApplicationDataRow[]>>;
+    applicationsToExclude: ApplicationDataRow[];
+    onAdd: () => void;
+    onClose: () => void;
+}
+
+const ApplicationsEntriesSelector: React.FC<ApplicationsEntriesSelectorProps> = ({
+                                        selectedApplications,
+                                        setSelectedApplications,
+                                        applicationsToExclude,
+                                        onAdd,
+                                        onClose,
+                                                                                 }) => {
+    const [applications, setApplications] = useState<ApplicationDataRow[]>([]);
+    const [selectAll, setSelectAll] = useState<boolean>(false);
+
+    useEffect(() => {
+        const fetchApplications = async () => {
+            try {
+                const data = await ManagmentServiceApiInstance.getApplications();
```

Comment:
Consider adding reportId to getApplications method of the ManagementServiceApiInstance

Reply:
good point